### DISCUSSION TITLE
How to patch SymbolsOnly font?

### DISCUSSION THREAD

**Discussion by BEST8OY** [ryanoasis/nerd-fonts#1751] [General]
Hello
I want to patch this font but I get an error message
```zsh
$ fontforge --script ./font-patcher --name SymbC --custom xxx.ttf ../../font/SymbolsNerdFont-Regular.ttf  
Copyright (c) 2000-2024. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Version: 20230101
 Based on sources from 2024-04-05 14:41 UTC-ML-D-GDK3.
Nerd Fonts Patcher v3.3.0 (4.16.1) (ff 20230101)
The glyph named heart is mapped to U+F08D0.
  But its name indicates it should be mapped to U+2665.
CRITICAL: Can not detect sane font width
```
I would appreciate any kind of help
Thanks.

**Comment by Finii** [comment]
The patcher expects a font with latin letters. It calculates a likely "cell size" by the size of specific latin letters and weights them in some way. If there are no letters (A-Z and a-z) it can not compute the "cell size" und thus not create a Nerd Font.

An exception is when the input font is completely empty. In that case the script decides on the "cell size" by whole-font properties.

A solution for you would be to take the (empty) SymbolsOnly input font (from under `/src/unpatched-fonts/NerdFontSymbolsOnly/` and patch that with `--complete --vari` and at the same time add your own icons with `--custom ...`.

Edit: Use this as input font https://github.com/ryanoasis/nerd-fonts/blob/master/src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbolsNerdFontBlank.sfd

**Reply by BEST8OY** [reply]
Yes
Thank you. It worked really well.

During all the testes I've done I figured that `--custom` when used with `--variable-width-glyphs` it messes with the bearings but when `--variable-width-glyphs` is used with the internal glyphs option of `font-patcher` it doesn't cause this issue!
So,
```-c  --variable-width-glyphs``` everything all right!
```-c --custom --variable-width-glyphs``` messes with the bearings!

Why is that? Is there any plan for it to get fixed in future?

**Reply by Finii** [reply]
I guess I do not really understand all your problems, but then probably I would need your `xxxt.ttf` to really examine it, and some example invocations that went well or not well; so that I could really reproduce the issues.

In principle `--custom` is not a core feature of the patcher and the support is rather ... limited. I used it once for a project (https://github.com/adam7/delugia-code) but that has a specifically hand-crafted custom font.

If I could understand the problem it can for sure be fixed. Not sure if that (fixing) is needed, noone ever complained so either it is ok or it's never used ;-)

**Reply by BEST8OY** [reply]
The font is Segoe Fluent Icons by Microsoft.
I installed it from AUR through a package.(V1.22)
The one in Microsoft website is v1.0
I thought maybe it's not okay to share it! (License stuff; I don't know much)
But here we go
~~ttf.zip~~
If it's not okay, please do tell me to remove it or if it's possible do so yourself.
Thanks.

**Reply by Finii** [reply]
As per their license we/you can not put it up here, neither modified or unmodified:

> ﻿You may use the Segoe and icon fonts, or glyphs included in this file (“Software”) solely to design, develop and test your programs that run on a Microsoft Platform, a Microsoft Platform includes but is not limited to any hardware or software product or service branded by trademark, trade dress, copyright or some other recognized means, as a product or service of Microsoft. **_This license does not grant you the right to distribute_** or sublicense all or part of the Software to any third party.  By using the Software, you agree to these terms. If you do not agree to these terms, do not use the Software.

_emphasis mine_
